src/include/sof/platform.h

```diff
@@ -54,6 +54,24 @@ int platform_context_save(struct sof *sof);
 
 /** @}*/
 
+#define HOST_TO_LOCAL 0
+#define LOCAL_TO_HOST 1
+
+static inline void convert_addr(int type, uint32_t *addr)
+{
+#ifdef CONFIG_IMX8ULP
+	switch (type) {
+	case HOST_TO_LOCAL:
+		*addr = SDRAM0_BASE + (*addr - MEM_RESERVED);
+		break;
+	case LOCAL_TO_HOST:
+		*addr = MEM_RESERVED + (*addr - SDRAM0_BASE);
+		break;
+	};
+#endif
+}
```

Comment:
seems like too many conditions to me. Functions like
```
void fn(bool mode)
{
	if (mode)
		do_one_thing();
	else
		do_another_thing();
}
```
which are then mostly called as `fn(true)` or `fn(false)` seem counterproductive to me - just call the `do_x_thing()` directly! Which here would mean - I'd define two distinct functions or macros `convert_local_to_host()` and `convert_host_to_local()` and just call them directly